src/openvic-simulation/map/MapInstance.cpp

```diff
@@ -128,41 +128,44 @@ bool MapInstance::apply_history_to_provinces(
 	return ret;
 }
 
-void MapInstance::update_modifier_sums(Date today, StaticModifierCache const& static_modifier_cache) {
+void MapInstance::update_modifier_sums(const Date today, StaticModifierCache const& static_modifier_cache) {
 	for (ProvinceInstance& province : province_instances.get_items()) {
 		province.update_modifier_sum(today, static_modifier_cache);
 	}
 }
 
-void MapInstance::update_gamestate(Date today, DefineManager const& define_manager) {
-	for (ProvinceInstance& province : province_instances.get_items()) {
-		province.update_gamestate(today, define_manager);
-	}
-	state_manager.update_gamestate();
-
-	// Update population stats
+void MapInstance::update_gamestate(const Date today, DefineManager const& define_manager) {
 	highest_province_population = 0;
 	total_map_population = 0;
+	
+	for (ProvinceInstance& province : province_instances.get_items()) {
+		province.update_gamestate(today, define_manager);
 
-	for (ProvinceInstance const& province : province_instances.get_items()) {
+		// Update population stats
 		const Pop::pop_size_t province_population = province.get_total_population();
-
 		if (highest_province_population < province_population) {
 			highest_province_population = province_population;
 		}
 
 		total_map_population += province_population;
 	}
+	state_manager.update_gamestate();
 }
 
-void MapInstance::tick(Date today) {
+void MapInstance::map_tick(const Date today, ModifierEffectCache const& modifier_effect_cache) {
 	for (ProvinceInstance& province : province_instances.get_items()) {
-		province.tick(today);
+		province.province_tick(today, modifier_effect_cache);
 	}
 }
 
-void MapInstance::initialise_for_new_game(ModifierEffectCache const& modifier_effect_cache){
+void MapInstance::initialise_for_new_game(
+	const Date today,
+	DefineManager const& define_manager,
+	ModifierEffectCache const& modifier_effect_cache
+) {
+	update_gamestate(today, define_manager);
 	for (ProvinceInstance& province : province_instances.get_items()) {
-		province.initialise_for_new_game(modifier_effect_cache);
+		province.initialise_rgo(modifier_effect_cache);
 	}
+	state_manager.tick(today, modifier_effect_cache);
```

Comment:
From my understanding, this function call ultimate results in every province belonging to a state being ticked?
- Why not just call `province.tick(...)` in the loop above after calling `initialise_rgo`?
- If the ticks must happen after all RGOs are intialised, why not just use the `MapInstance`'s tick function? Yes it'll also tick water provinces, but that's likely more than offset by the memory operations of having to loop through each state set and state, in particular because that will result in provinces being ticked in an irregular order rather than going through them in the order they're stored in memory like `MapInstance`'s tick function does.
- If we really do need to do this by state, then these starting-province-tick functions need different names as we'll likely have proper state tick functions eventually for processing state-level stuff like factories and tension, which won't tick individual provinces.

Reply:
I think I did that to avoid ticking unowned provinces. I change simply add the province tick in the loop above. It will exit early if there is no rgo.